spec/expression.dd

```diff
@@ -1695,37 +1695,15 @@ $(GNAME ArrayMemberInitialization):
         ---
 
     $(P By default, an array literal is typed as a dynamic array, but the element
-        count is known at compile time. So all array literals can be
-        implicitly converted to static array types. Slicing a dynamic array with
-        statically known bounds also allows conversion to a static array.)
+        count is known at compile time. So an array literal can be
+        implicitly converted to a static array of the same length.)
 
-        $(SPEC_RUNNABLE_EXAMPLE_RUN
         -------------
-        void foo(long[2] a)
-        {
-            assert(a == [2, 3]);
-        }
-        void bar(ref long[2] a)
-        {
-            assert(a == [2, 3]);
-            a = [4, 5];
-            assert(a == [4, 5]);
-        }
-
-        void main()
-        {
-            long[] arr = [1, 2, 3];
-
-            foo(arr[1 .. 3]);
-            assert(arr == [1, 2, 3]);
-
-            bar(arr[1 .. 3]);
-            assert(arr == [1, 4, 5]);
-
-            //foo(arr[0 .. 3]); // cannot match length
-        }
+        int[2] sa = [1, 2];
         -------------
-        )
+
+    $(NOTE Slicing a dynamic array with statically known bounds also
+        $(RELATIVE_LINK2 slice_to_static_array, allows conversion) to a static array.)
```

Comment:
Implicit conversion is possible when the resulting *size* is known at compile time.  DMD recognizes e.g. this pattern:

```d
void foo(int[] array, int i)
{
    int[2] part = array[i .. i + 2];
    assert(part == [ 2, 3 ]);
}

void main()
{
    int[] arr = [ 1, 2, 3, 4 ];
    foo(arr, 1);
}
```

Reply:
BTW, I've added an example of this in #3231.